src/main/java/org/jabref/gui/DialogService.java

```diff
@@ -168,6 +168,14 @@ boolean showConfirmationDialogWithOptOutAndWait(String title, String content,
      */
     void showCustomDialog(BaseDialog<?> dialog);
 
+    /**
+     * This will create and display a new dialog showing a custom {@link DialogPane}
+     * and using custom {@link ButtonType}s.
+     *
+     *
+     */
+    Optional<ButtonType> showNonModalCustomDialogAndWait(String title, DialogPane contentPane, ButtonType... buttonTypes);
```

Comment:
Maybe add the modality as a new optional enum to the already existing showCustomDialogAndWait method?

Reply:
As this addition is doubted, in the current state of implementation of this feature this is not used anymore and we want to finish this PR I removed it. Maybe we can add a modal enum in a later PR when we need it again.